lte/gateway/c/oai/tasks/sgw/sgw_handlers.c

```diff
@@ -685,145 +726,176 @@ int sgw_handle_sgi_endpoint_deleted(
-      eps_bearer_ctxt_p->enb_teid_S1u =
-          modify_bearer_pP->bearer_contexts_to_be_modified.bearer_contexts[0]
-              .s1_eNB_fteid.teid;
-      {
-        itti_sgi_update_end_point_response_t sgi_update_end_point_resp = {0};
-
-        sgi_update_end_point_resp.context_teid = modify_bearer_pP->teid;
-        sgi_update_end_point_resp.sgw_S1u_teid =
-            eps_bearer_ctxt_p->s_gw_teid_S1u_S12_S4_up;
-        sgi_update_end_point_resp.enb_S1u_teid =
-            eps_bearer_ctxt_p->enb_teid_S1u;
-        sgi_update_end_point_resp.eps_bearer_id =
+    }  // for loop
+    for (idx = 0;
+         idx <
+         modify_bearer_pP->bearer_contexts_to_be_removed.num_bearer_context;
+         idx++) {
+      eps_bearer_ctxt_p = sgw_cm_get_eps_bearer_entry(
+          &bearer_ctxt_info_p->sgw_eps_bearer_context_information
+               .pdn_connection,
+          modify_bearer_pP->bearer_contexts_to_be_removed.bearer_contexts[idx]
+              .eps_bearer_id);
+      if (eps_bearer_ctxt_p) {
+        sgi_update_end_point_resp.bearer_contexts_to_be_removed[sgi_rsp_idx++] =
             eps_bearer_ctxt_p->eps_bearer_id;
-        sgi_update_end_point_resp.status = 0x00;
-        sgw_handle_sgi_endpoint_updated(&sgi_update_end_point_resp, imsi64);
+        sgi_update_end_point_resp.num_bearers_removed++;
       }
     }
-  } else {
-    message_p =
-        itti_alloc_new_message(TASK_SPGW_APP, S11_MODIFY_BEARER_RESPONSE);
-
-    if (!message_p) {
-      OAILOG_FUNC_RETURN(LOG_SPGW_APP, RETURNerror);
+  } else {  // bearer_ctxt_info_p not found
+    rv = send_mbr_failure(modify_bearer_pP, imsi64);
+    if (rv != RETURNok) {
+      OAILOG_ERROR(
+          LOG_SPGW_APP,
+          "Error in sending modify bearer response to MME App for the failed "
+          "bearers, teid" TEID_FMT "\n",
+          modify_bearer_pP->teid);
     }
-
-    modify_response_p = &message_p->ittiMsg.s11_modify_bearer_response;
-    modify_response_p->bearer_contexts_marked_for_removal.bearer_contexts[0]
-        .eps_bearer_id =
-        modify_bearer_pP->bearer_contexts_to_be_modified.bearer_contexts[0]
-            .eps_bearer_id;
-    modify_response_p->bearer_contexts_marked_for_removal.bearer_contexts[0]
-        .cause.cause_value = CONTEXT_NOT_FOUND;
-    modify_response_p->bearer_contexts_marked_for_removal.num_bearer_context +=
-        1;
-    modify_response_p->cause.cause_value = CONTEXT_NOT_FOUND;
-    modify_response_p->trxn              = modify_bearer_pP->trxn;
-    OAILOG_DEBUG_UE(
-        LOG_SPGW_APP, imsi64,
-        "Rx MODIFY_BEARER_REQUEST, teid " TEID_FMT " CONTEXT_NOT_FOUND\n",
-        modify_bearer_pP->teid);
-
-    message_p->ittiMsgHeader.imsi = imsi64;
-    rv = send_msg_to_task(&spgw_app_task_zmq_ctx, TASK_MME, message_p);
-
     OAILOG_FUNC_RETURN(LOG_SPGW_APP, rv);
   }
-
-  OAILOG_FUNC_RETURN(LOG_SPGW_APP, RETURNerror);
+  sgw_handle_sgi_endpoint_updated(&sgi_update_end_point_resp, imsi64);
```

Comment:
move this to the end of the `if (bearer_ctxt_info_p) { ` to make it more readable as this is only supposed to be done when the if statement is true.